craft_providers/lxd/lxc.py

```diff
@@ -995,3 +1032,85 @@ def stop(
                 brief=f"Failed to stop {instance_name!r}.",
                 details=errors.details_from_called_process_error(error),
             ) from error
+
+    def check_instance_status(
+        self,
+        *,
+        instance_name: str,
+        project: str = "default",
+        remote: str = "local",
+    ) -> None:
+        """Check build status of instance."""
+        build_status: Optional[str] = None
+        instance_info: Dict[str, Any] = {"status": ""}
+        retry_count: int = 0
+
+        # Approx max wait time is 2 hours
+        while retry_count < 2400:
```

Comment:
I would prefer a more standard time-based timeout and loop for two reasons:
1. readability - 2,400 retries is hard to comprehend. It took me a while to wrap my head around how long a `retry_count` is (like on the line `if retry_count == 100:`), since I have to divide 2,400 retries by approximately 2 hours
2. predictability - Timeouts are a bit more predictable.  I can imagine the actual timeout of 2,400 retries varying from system to system.

Reply:
I thinking that too, but I move to time + count based.